src/GridFS/ReadableStream.php

```diff
@@ -176,6 +177,28 @@ public function readBytes(int $length): string
         return $data;
     }
 
+    /**
+     * Rename all revisions of the file.
+     *
+     * @return int The number of revisions renamed
+     */
+    public function rename(string $newFilename): int
+    {
+        /** @var Document[] $revisions */
+        $revisions = $this->collectionWrapper->findFiles(
+            ['filename' => $this->file->filename],
+            ['typeMap' => ['root' => 'bson'], 'projection' => ['_id' => 1]],
+        );
+        $count = 0;
+
+        foreach ($revisions as $revision) {
+            $this->collectionWrapper->updateFilenameForId($revision->get('_id'), $newFilename);
+            $count++;
+        }
+
+        return $count;
```

Comment:
It's explicitly written in the [spec](https://github.com/mongodb/specifications/blob/master/source/gridfs/gridfs-spec.rst#renaming-stored-files) to perform the update one-by-one.

> Drivers construct and execute an update_one command on the files collection using `{ _id: @id }` as the filter and `{ $set : { filename : "new_filename" } }` as the update parameter.

Reply:
This may be worth bringing up with the wider DBX team, as that GridFS spec language definitely dates back a while and could stand to be updated. Re-reading it, it's clear to me that one of the priorities was raising an error if a file doesn't exist for a given `_id`; however, I see no technical reason why an `updateMany()` with all `_id` values expressed in a single filter (e.g. using `$in`) couldn't be used, provided the method also asserted that the expected number of documents were modified by the bulk update.

What you certainly don't want to do is attempt to update many documents based on the filename directly, since that could end up hitting additional file records that we didn't expect. That's certainly not happening here, but I wanted to point it out regardless.